### DISCUSSION TITLE
SplashScreen only works with Window, but not winuiex:WindowEx?

### DISCUSSION THREAD

**Discussion by Asgardi** [dotMorten/WinUIEx#79] [Q&A]
Would be nice of the type was not forced to work with vanilla Window type, but WindowEx type instead (or both?).

**Comment by dotMorten** [comment]
Could you explain what you mean by that? I'm not quite following.

**Comment by Asgardi** [comment]
You create a new splash by giving your main window type: 
var splash = new SplashScreen(typeof(MainWindow));

Then you register to the Completed event like this to get the activated MainWindow:
splash.Completed += (s, e) => m_window = e;

The problem is that "e" is of type "Window", but my MainWindow is of type "WindowEx".

**Comment by dotMorten** [comment]
`WindowEx` _is_ a `Window`. So you can just cast it back to `WindowEx`.

`splash.Completed += (s, e) => m_window = (WindowEx)e;`

**Reply by Asgardi** [reply]
Well duh... Sorry for the wrong alarm :D

**Reply by dotMorten** [reply]
No worries - We all have our days :D
